### PR TITLE
Emme network scripts

### PR DESCRIPTION
**Pull request by hsl-petrhaj**
Here I tried to address all the recent issues touching the network validation. I typically started by writing the test and then implemented the solution in network_validation.py. I am also publishing the tests because I think they might be useful in the future.
Some small reogranization in the folder structure is also proposed.

### CODE REVIEW

File: Scripts/utils/validate_network.py [hunk 1]
```diff
@@ -61,8 +61,26 @@ def validate(network, fares=None):
         intervals += param.official_node_numbers[modes]
     unofficial_nodes = set()
     for link in network.links():
+        if len(link.modes) == 0:
```
Comment (zptro) [inline Scripts/utils/validate_network.py:64]: This could be written as
```suggestion
        if not link.modes:
```

**Review by zptro — DISMISSED** [review]
Very nice commit. Thank you for this!

File: Scripts/utils/validate_network.py (hunk 1 above)
Comment (hsl-petrhaj) [reply Scripts/utils/validate_network.py]: This should be in some "extreme Python" lessons :D

**Review by johpiip — DISMISSED** [review]
Great error messages! :)

**Comment by hsl-petrhaj** [comment]
Note: even our test network had some 999 links, I corrected these for some suitable close enough link type.

File: Scripts/utils/validate_network.py [hunk 2]
```diff
@@ -99,7 +107,14 @@ def validate(network, fares=None):
...
                     log.error(msg)
                     raise ValueError(msg)
+                
+        if link.i_node.is_centroid and link.j_node.is_centroid:
+            msg = "Link {} is leading directly from centroid node {} to centroid node {}. This is not allowed.".format(link.id,link.i_node.number,link.j_node.number)
+            log.error(msg)
+            raise ValueError(msg)
+        
         for node in (link.i_node, link.j_node):
+
```
Comment (zptro) [inline Scripts/utils/validate_network.py:117]: Why this extra row here?

**Review by zptro — APPROVED** [review]

**Pull request merged**